pkg/tsdb/azuremonitor/azcredentials/credentials.go

```diff
@@ -21,10 +22,19 @@ type AzureClientSecretCredentials struct {
 	ClientSecret string
 }
 
+type AzureUserIdentityCredentials struct {
+	TokenEndpoint string // token endpoint to retrieve the token from
+	AuthHeader    string // exact Authorization header to be used to talk to the token endpoint
```

Comment:
These settings should not be a part of the credential object, it's not configuration of the credential (e.g. user doesn't provide this input).

This is implementation-specific configuration of the AzureTokenProvider and it can access these settings directly.